becquerel/core/fitting.py

```diff
@@ -338,12 +340,45 @@ def __str__(self):
             '        x: {}\n'.format(self.x) +
             '        y: {}\n'.format(self.y) +
             '    y_unc: {}\n'.format(self.y_unc) +
+            '    xmode: {}\n'.format(self.xmode) +
+            '    ymode: {}\n'.format(self.ymode) +
             '       dx: {}\n'.format(self.dx) +
             '      roi: {}'.format(self.roi)
         )
 
     __repr__ = __str__
 
+    @classmethod
+    def from_spectrum(cls, model, spec, xmode, ymode, roi=None):
```

Comment:
From meeting: also connect fitting to `Spectrum` with `Spectrum.fit`

Reply:
Done!